File: src/main/java/com/cuupa/dms/database/user/DbUser.java
package com.cuupa.dms.database.user;

import javax.persistence.*;

@Entity
@Table(name = "dmsusers")
public class DbUser {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    private String username;

    private String password;

    private String salt;

    private String firstname;

    private String lastname;

    private String accesstoken;

    public DbUser(String username, String password, String salt, String firstname, String lastname) {
        this.username = username;
        this.password = password;
        this.salt = salt;
        this.firstname = firstname;
        this.lastname = lastname;
    }

    public DbUser() {
    }

    public Integer getId() {
        return id;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public String getAccesstoken() {
        return accesstoken;
    }

    public void setAccesstoken(String accesstoken) {
        this.accesstoken = accesstoken;
    }

    @Override
    public String toString() {
        return "User{" +
               "id=" +
               id +
               ", username='" +
               username +
               '\'' +
               ", firstname='" +
               firstname +
               '\'' +
               ", surname='" +
               lastname +
               '\'' +
               '}';
    }

    public String getSalt() {
        return salt;
    }
}

File: src/main/java/com/cuupa/dms/database/user/DbUser.kt
package com.cuupa.dms.database.user

import javax.persistence.*

@Entity
@Table(name = "dmsusers")
class DbUser {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Int? = null
        private set
    lateinit var username: String
    lateinit var password: String
    lateinit var salt: String
    var firstname: String? = null
    var lastname: String? = null
    lateinit var accesstoken: String
    var isConfirmed = false

    constructor(id: Int, username: String, password: String, salt: String, firstname: String?, lastname: String?,
                accesstoken: String, confirmed: Boolean) {
        this.id = id
        this.username = username
        this.password = password
        this.salt = salt
        this.firstname = firstname
        this.lastname = lastname
        this.accesstoken = accesstoken
        isConfirmed = confirmed
    }

    constructor(username: String, password: String, salt: String, firstname: String?, lastname: String?,
                accesstoken: String, confirmed: Boolean) {
        this.username = username
        this.password = password
        this.salt = salt
        this.firstname = firstname
        this.lastname = lastname
        this.accesstoken = accesstoken
        isConfirmed = confirmed
    }

    constructor()

    override fun toString(): String {
        return "User{" + "id=" + id + ", username='" + username + '\'' + ", firstname='" +
                firstname +
                '\'' +
                ", lastname='" +
                lastname +
                '\'' +
                '}'
    }

}